src/StreamAPI.php

```diff
@@ -0,0 +1,464 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ToshY\BunnyNet;
+
+use Psr\Http\Client\ClientExceptionInterface;
+use ToshY\BunnyNet\Client\BunnyClient;
+use ToshY\BunnyNet\Enum\Host;
+use ToshY\BunnyNet\Exception\FileDoesNotExistException;
+use ToshY\BunnyNet\Helper\BodyContentHelper;
+use ToshY\BunnyNet\Model\API\Stream\ManageCollections\CreateCollection;
+use ToshY\BunnyNet\Model\API\Stream\ManageCollections\DeleteCollection;
+use ToshY\BunnyNet\Model\API\Stream\ManageCollections\GetCollection;
+use ToshY\BunnyNet\Model\API\Stream\ManageCollections\ListCollections;
+use ToshY\BunnyNet\Model\API\Stream\ManageCollections\UpdateCollection;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\AddCaption;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\CreateVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\DeleteCaption;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\DeleteVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\FetchVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\GetVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\GetVideoHeatmap;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\ListVideos;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\ListVideoStatistics;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\ReEncodeVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\SetThumbnail;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\UpdateVideo;
+use ToshY\BunnyNet\Model\API\Stream\ManageVideos\UploadVideo;
+use ToshY\BunnyNet\Model\Client\Interface\BunnyClientResponseInterface;
+use ToshY\BunnyNet\Validator\ParameterValidator;
+
+class StreamAPI
+{
+    /**
+     * @param string $apiKey
+     * @param BunnyClient $client
+     */
+    public function __construct(
+        protected readonly string $apiKey,
+        protected readonly BunnyClient $client,
+    ) {
+        $this->client
+            ->setApiKey($this->apiKey)
+            ->setBaseUrl(Host::STREAM_ENDPOINT);
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @param string $collectionId
+     * @return BunnyClientResponseInterface
+     * @param int $libraryId
+     */
+    public function getCollection(int $libraryId, string $collectionId): BunnyClientResponseInterface
```

Comment:
- [ ] Unsure if 2 arguments should be multilined or just kept on single line. Need to think about it, cuz on single line a trailing comma looks kind of strange.